renku/core/management/datasets.py

```diff
@@ -383,12 +385,29 @@ def set_dataset_images(self, dataset, images, safe_image_paths=[]):
             if existing:
                 dataset.images.append(existing)
                 continue
-
+            image_type = None
             if urlparse(content_url).netloc:
                 # NOTE: absolute url
-                dataset.images.append(ImageObject(content_url, position, id=ImageObject.generate_id(dataset, position)))
-                images_updated = True
-                continue
+                if not img.get("mirror_locally", False):
+                    dataset.images.append(
+                        ImageObject(content_url, position, id=ImageObject.generate_id(dataset, position))
+                    )
+                    images_updated = True
+                    continue
+
+                # NOTE: mirror the image locally
+                try:
+                    path, _ = urlretrieve(content_url)
+                except error.URLError as e:
+                    raise errors.DatasetImageError(f"Dataset image with url {content_url} couldn't be mirrored") from e
+
+                image_type = imghdr.what(path)
+                if not image_type:
+                    raise errors.DatasetImageError(f"Dataset image with url {content_url} is not a valid image type")
```

Comment:
Should we raise here? It seems that `imghdr` does not work all the time: https://stackoverflow.com/questions/36870661/imghdr-python-cant-detec-type-of-some-images-image-extension. This might cause unnecessary bug reports in future and I don't really see any value in doing this. UI can do the check as it needs to do it anyway for image URLs.
Maybe just use `imghdr` to detect the extension.